### PR TITLE
CATTY-16 Play Sound And Wait Brick

### PR DESCRIPTION
**Pull request by Wullschi**
- Added the new brick "Play sound and wait" which pauses a script until the sound has finished playing.
- Added Three integration tests (with three new project files) to test the new functionality.
- Refactored some existing audio player unit tests.
- Minor refactoring of the AudioPlayer class.
- Minor refactoring of some test classes
- Adapted the cartfile to load most recent version of AudioKit

### CODE REVIEW

**Comment by Wullschi** [comment]
Test project: https://share.catrob.at/app/project/6d357a58-1a08-11ea-8f2b-000c292a0f49

File: src/Catty/DataModel/Bricks/Sound/PlaySoundAndWaitBrick.m [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+#pragma mark - Description
+- (NSString*)description
+{
+    return [NSString stringWithFormat:@"PlaySoundAndWait (File Name: %@)", self.sound.fileName];
+}
+
+- (BOOL)isEqualToBrick:(Brick*)brick
+{
+    if(![self.sound isEqualToSound:((PlaySoundAndWaitBrick*)brick).sound])
```
Comment (m-herold) [inline src/Catty/DataModel/Bricks/Sound/PlaySoundAndWaitBrick.m]: Could you please double check this line. What happens if the Brick is not of type `PlaySoundAndWaitBrick`?

File: src/Cartfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ github "TTTAttributedLabel/TTTAttributedLabel" ~> 2.0.0
 github "maxep/MXSegmentedPager" ~> 3.3.0
 github "onevcat/Kingfisher" ~> 4.0
 github "venmo/DVR"
-github "audiokit/audiokit" "v4.9.2"
+github "audiokit/audiokit" "v4.9.3"
```
Comment (m-herold) [inline src/Cartfile]: Nice!

File: src/Catty/DataModel/Bricks/Sound/PlaySoundAndWaitBrick.m (hunk 1 above)
Comment (Wullschi) [reply src/Catty/DataModel/Bricks/Sound/PlaySoundAndWaitBrick.m]: Thanks, I'll add a type check to this method.

Comment (m-herold) [reply src/Catty/DataModel/Bricks/Sound/PlaySoundAndWaitBrick.m]: Perfect, thanks!

**Pull request merged**